packages/common/src/adapters/notification.ts

```diff
@@ -0,0 +1,500 @@
+        entityType = getDiscoveryEntityType(data)
+        return decodeHashId(data.userId)
+      })
+      .filter(removeNullable)
+    return {
+      type: NotificationType.RepostOfRepost,
+      userIds,
+      entityId,
+      entityType,
+      ...formatBaseNotification(notification)
+    }
+  } else if (notification.type === 'save_of_repost') {
+    let entityId = 0
+    let entityType = Entity.Track
+    const userIds = notification.actions
+      .map((action) => {
+        const data = action.data
+        entityId = decodeHashId(data.saveOfRepostItemId) as number
+        entityType = getDiscoveryEntityType(data)
+        return decodeHashId(data.userId)
+      })
+      .filter(removeNullable)
+    return {
+      type: NotificationType.FavoriteOfRepost,
+      userIds,
+      entityId,
+      entityType,
+      ...formatBaseNotification(notification)
+    }
+  } else if (notification.type === 'usdc_purchase_seller') {
+    let entityId = 0
+    let entityType = Entity.Track
+    let amount = '' as StringUSDC
+    let extraAmount = '' as StringUSDC
+    const userIds = notification.actions
+      .map((action) => {
+        const data = action.data
+        entityId = decodeHashId(data.contentId) as number
+        entityType = data.contentType === 'track' ? Entity.Track : Entity.Album
+        amount = data.amount as StringUSDC
+        extraAmount = data.extraAmount as StringUSDC
+        return decodeHashId(data.buyerUserId)
+      })
+      .filter(removeNullable)
+    return {
+      type: NotificationType.USDCPurchaseSeller,
+      userIds,
+      entityId,
+      entityType,
+      amount,
+      extraAmount,
+      ...formatBaseNotification(notification)
+    }
+  } else if (notification.type === 'usdc_purchase_buyer') {
+    let entityId = 0
+    let entityType = Entity.Track
+    const userIds = notification.actions
+      .map((action) => {
+        const data = action.data
+        entityId = decodeHashId(data.contentId) as number
+        entityType = data.contentType === 'track' ? Entity.Track : Entity.Album
+        return decodeHashId(data.sellerUserId)
+      })
+      .filter(removeNullable)
+    return {
+      type: NotificationType.USDCPurchaseBuyer,
+      userIds,
+      entityId,
+      entityType,
+      ...formatBaseNotification(notification)
+    }
+  } else if (notification.type === 'request_manager') {
+    const data = notification.actions[0].data
+
+    return {
+      type: NotificationType.RequestManager,
+      userId: decodeHashId(data.userId)!,
+      ...formatBaseNotification(notification)
+    }
+  } else if (notification.type === 'approve_manager_request') {
+    const data = notification.actions[0].data
+
+    return {
+      type: NotificationType.ApproveManagerRequest,
+      userId: decodeHashId(data.granteeUserId)!,
+      ...formatBaseNotification(notification)
+    }
+  } else {
+    console.error('Notification does not match an expected type.')
```

Comment:
Wondering if this should throw or something? What happens if we let an unprocessed entity through?

Reply:
SDK throws before we get here as we exhaust all possible values for the enum now! I made this a switch/case instead of if/else so that we get typescript checking to ensure we exhaust all notif types here